driver/normalizer/annotation.go

```diff
@@ -38,5 +40,461 @@ var Code []CodeTransformer // legacy stage, will be deprecated
 
 // Annotations is a list of individual transformations to annotate a native AST with roles.
 var Annotations = []Mapping{
+
+	// Misc
 	AnnotateType("internal-type", nil, role.Incomplete),
+	AnnotateType("CompilationUnit", nil, role.File, role.Module),
+	AnnotateType("Block", nil, role.Block),
+	AnnotateType("LockStatement", nil, role. Statement, role.Block, role.Incomplete),
+	AnnotateType("CheckedStatement_UncheckedStatement", nil, role.Block, role.Statement, role.Incomplete),
+	AnnotateType("CheckedExpression_UncheckedExpression", nil, role.Block, role.Expression, role.Incomplete),
+	AnnotateType("UnsafeStatement", nil, role.Block, role.Statement, role.Incomplete),
+	AnnotateType("FixedStatement", nil, role.Declaration, role.Incomplete),
+	AnnotateType("NamespaceDeclaration", nil, role.Block, role.Scope),
+	AnnotateType("EventFieldDeclaration", nil, role.Declaration, role.Variable),
+	AnnotateType("EventDeclaration", nil, role.Declaration, role.Variable),
+	AnnotateType("ArrayType", nil, role.List, role.Incomplete),
+	AnnotateType("ArrayRankSpecifier", nil, role.List, role.Incomplete),
+	AnnotateType("BracketedArgumentList", nil, role.List, role.Value, role.Incomplete), // i in someaArray[i]
+	AnnotateType("OmittedArraySizeExpression", nil, role.List, role.Expression, role.Incomplete),
+	AnnotateType("ArrayCreationExpression", nil, role.List, role.Expression, role.Instance, role.Incomplete),
+	AnnotateType("ImplicitArrayCreationExpression", nil, role.List, role.Expression, role.Value, role.Incomplete),
+	AnnotateType("StackAllocArrayCreationExpression", nil, role.List, role.Expression, role.Instance, role.Incomplete),
+	AnnotateType("ArrayInitializerExpression", nil, role.List, role.Expression, role.Instance, role.Incomplete),
+	AnnotateType("ElementAccessExpression", nil, role.List, role.Value, role.Incomplete),
+	AnnotateType("CastExpression", nil, role.Expression, role.Incomplete),
+	AnnotateType("PredefinedType", nil, role.Incomplete, role.Declaration, role.Variable),
```

Comment:
This doesn't look right to me. If I'm not mistaken, `PredefinedType` wraps an identifier to signal that this is a primitive type defined by the language. Here it includes a `Declaration` and `Variable` roles, instead of `Type`.